core/lib/config/src/configs/external_price_api_client.rs

```diff
@@ -4,16 +4,21 @@ use serde::Deserialize;
 
 pub const DEFAULT_TIMEOUT_MS: u64 = 10_000;
 
+pub const DEFAULT_FORCED_NEXT_VALUE_FLUCTUATION: u32 = 3;
+
 #[derive(Debug, Clone, PartialEq, Deserialize)]
 pub struct ForcedPriceClientConfig {
     /// Forced conversion ratio
     pub numerator: Option<u64>,
     pub denominator: Option<u64>,
-    /// Forced fluctuation. It defines how much percent numerator /
-    /// denominator should fluctuate from their forced values. If it's None or 0, then ForcedPriceClient
-    /// will return the same quote every time it's called. Otherwise, ForcedPriceClient will return
-    /// forced_quote +/- forced_fluctuation % from its values.
+    /// Forced fluctuation. It defines how much percent the ratio should fluctuate from its forced
+    /// value. If it's None or 0, then the ForcedPriceClient will return the same quote every time
+    /// it's called. Otherwise, ForcedPriceClient will return quote with numerator +/- fluctuation %.
     pub fluctuation: Option<u32>,
+    /// In order to smooth out fluctuation, consecutive values returned by forced client will not
+    /// differ more than next_value_fluctuation percent. If it's None, a default of 3% will be applied.
+    #[serde(default = "ExternalPriceApiClientConfig::default_forced_next_value_fluctuation")]
```

Comment:
I have done in like this because we still have some compatibility with serde based config parsing (from env vars?). I have separately implemented the defaults in protobuf config. What would you want to me to do if I want this to be optional and have a default? _I think we should overhaul how configs are handled, so there is one code for parsing only... Maybe we should remove serde?_

Reply:
It's done like I did it in other configs eg. prover_job_monitor

Reply:
As you are vactioning, I'm closing this.